tests_integ/helpers/base_test.py

```diff
@@ -0,0 +1,370 @@
+        except ClientError as error:
+            LOG.error("Upload of file %s to bucket %s failed", current_file_name, cls.s3_bucket_name, exc_info=error)
+            cls._clean_bucket()
+            raise error
+
+    @classmethod
+    def _get_s3_uri(cls, file_name, uri_type):
+        if uri_type == "s3":
+            return "s3://{}/{}".format(cls.s3_bucket_name, file_name)
+
+        if cls.my_region == "us-east-1":
+            return "https://s3.amazonaws.com/{}/{}".format(cls.s3_bucket_name, file_name)
+        if cls.my_region == "us-iso-east-1":
+            return "https://s3.us-iso-east-1.c2s.ic.gov/{}/{}".format(cls.s3_bucket_name, file_name)
+        if cls.my_region == "us-isob-east-1":
+            return "https://s3.us-isob-east-1.sc2s.sgov.gov/{}/{}".format(cls.s3_bucket_name, file_name)
+
+        return "https://s3-{}.amazonaws.com/{}/{}".format(cls.my_region, cls.s3_bucket_name, file_name)
+
+    def setUp(self):
+        self.cloudformation_client = boto3.client("cloudformation")
+        self.deployer = Deployer(self.cloudformation_client)
+
+    def tearDown(self):
+        self.cloudformation_client.delete_stack(StackName=self.stack_name)
+        if os.path.exists(self.output_file_path):
+            os.remove(self.output_file_path)
+        if os.path.exists(self.sub_input_file_path):
+            os.remove(self.sub_input_file_path)
+
+    def create_and_verify_stack(self, file_name, parameters=None):
+        """
+        Creates the Cloud Formation stack and verifies it against the expected
+        result
+
+        Parameters
+        ----------
+        file_name : string
+            Template file name
+        parameters : list
+            List of parameters
+        """
+        self.output_file_path = str(Path(self.output_dir, "cfn_" + file_name + ".yaml"))
+        self.expected_resource_path = str(Path(self.expected_dir, file_name + ".json"))
+        self.stack_name = STACK_NAME_PREFIX + file_name.replace("_", "-") + "-" + generate_suffix()
+
+        self._fill_template(file_name)
+        self.transform_template()
+        self.deploy_stack(parameters)
+        self.verify_stack()
+
+    def transform_template(self):
+        transform_template(self.sub_input_file_path, self.output_file_path)
+
+    def get_region(self):
+        return self.my_region
+
+    def get_s3_uri(self, file_name):
+        """
+        Returns the S3 URI of a resource file
+
+        Parameters
+        ----------
+        file_name : string
+            Resource file name
+        """
+        return FILE_TO_S3_URI_MAP[file_name]["uri"]
+
+    def get_code_key_s3_uri(self, code_key):
+        """
+        Returns the S3 URI of a code key for template replacement
+
+        Parameters
+        ----------
+        code_key : string
+            Template code key
+        """
+        return FILE_TO_S3_URI_MAP[CODE_KEY_TO_FILE_MAP[code_key]]["uri"]
```

Comment:
This also feels like it should be an instance variable that we seed, for flexibility - both values, arguably.

Reply:
Yes, I have updated it to use instance variables.